pkg/codegen/go/gen_program_expressions.go

```diff
@@ -96,7 +96,13 @@ func (g *generator) genAnonymousFunctionExpression(
 	body, temps := g.lowerExpression(expr.Body, retType)
 	g.genTempsMultiReturn(w, temps, retTypeName)
 
-	g.Fgenf(w, "return %v, nil", body)
+	// fromBase64 special case
+	if strings.Contains(fmt.Sprintf("%v", body), "fromBase64") {
```

Comment:
This seems like a very fragile check. Is there a different way we can do this?

Reply:
Unfortunately as an `AnonymousFunctionExpression`, we don't have an explicit name.  `fromBase64` is our internal function mapping used to prefix the body (i.e. `fromBase64(__convert(aksCredentials.kubeconfigs[0].value))` – would [centralizing all of these](https://github.com/pulumi/pulumi/blob/master/pkg/codegen/go/gen_program_expressions.go#L218-L336) into one place suffice?

Reply:
Somehow, `body` knows how to format itself to include the string `fromBase64`. Do you know where this string is stored in `body`?

Reply:
It should be defined [here](https://github.com/pulumi/pulumi/blob/aqiu/10726/pkg/codegen/pcl/functions.go#L296) in `functions.go`, referenced by the `[binder](https://github.com/pulumi/pulumi/blob/master/pkg/codegen/pcl/binder.go#L133-L135)`

Reply:
After https://github.com/pulumi/pulumi/pull/10775/commits/ea4fd10299604846b9bdfef20f64207bd6b9c1f3 landed, I ran the tests with some added debug printing. It's not a `AnonymousFunctionExpression`. Its a `FunctionCallExpression`, which makes sense. Testing on my machine, this works:
```suggestion
	if body, ok := body.(*model.FunctionCallExpression); ok && body.Name == "fromBase64" {
```